example/main.go

```diff
@@ -76,8 +75,7 @@ func main() {
 	// Setup webhooks
 	mutatingWebhook, err := builder.NewWebhookBuilder().
 		Name("mutating.k8s.io").
-		Type(types.WebhookTypeMutating).
-		Path("/mutating-pods").
+		Mutating().
```

Comment:
Maybe change the function names to `BeMutating` and `BeValidating`.

Reply:
IMHO changing methods names to `BeMutating` and `BeValidating` doesn't buy us much. And just call if `Mutating` is more concise.